MediaBrowser.Providers/TV/SeriesMetadataService.cs

```diff
@@ -67,6 +67,25 @@ protected override void MergeData(MetadataResult<Series> source, MetadataResult<
 
             var sourceItem = source.Item;
             var targetItem = target.Item;
+            var sourceSeasonNames = sourceItem.SeasonNames;
+            var targetSeasonNames = targetItem.SeasonNames;
+
+            if (replaceData || targetSeasonNames is null)
+            {
+                targetItem.SeasonNames = sourceSeasonNames;
+            }
+            else if (!(targetSeasonNames.Count == sourceSeasonNames.Count
+                && targetSeasonNames.Keys.All(sourceSeasonNames.ContainsKey)))
```

Comment:
I don't think much is gained by having this check (line 77 is fine). `Keys` creates a `KeyCollection` (a class), which you then enumerate with LINQ, so nothing is really saved. I'm also not sure the check is correct. You only want to set _new_ key-value pairs in targetItem, but you check whether all targetItem keys are in sourceItem. Shouldn't it be the other way around?

Reply:
The condition in itself is inverted, so I think the logic is correct.
What I wanted to achieve is this: If lengths are not the same and the series does not have a name for a season the new season names are added.

But as you pointed out, the `SeasonNames` is not persisted within the database, so it is lost on restart, I guess we can circumvent it by manually defining a getter, building the dictionary based on the season metadata.

Reply:
It works because if the count is the same, then they are either identical or not, but it's not "correct". It should've been `targetSeasonNames.Count != sourceSeasonNames.Count
			  || !sourceSeasonNames.Keys.All(targetSeasonNames.ContainsKey)`

Reply:
Right, the keys could be different if the length is the same.

Reply:
It's the same result in the end, but the intention somewhat differs.